src/sidebar/components/share-annotations-panel.js

```diff
@@ -59,55 +58,70 @@ function ShareAnnotationsPanel({ analytics, toastMessenger }) {
       title={panelTitle}
       panelName={uiConstants.PANEL_SHARE_ANNOTATIONS}
     >
-      {focusedGroup && mainFrame && (
+      {!sharingReady && (
```

Comment:
I hope the states are more clear now. There is one top-level branch now: "is shiz loaded enough yet?" and one secondary branch when the answer is "yes": "Do we have a valid share URI?"

Reply:
This makes sense to me. If it gets significantly more complex in future that it might help to factor out a helper component in the same module.